source/server-discovery-and-monitoring/server-discovery-and-monitoring-logging-and-monitoring.rst

```diff
@@ -138,7 +138,7 @@ Events that MUST be published (with their conditions) are as follows.
    * - ``TopologyClosedEvent``
      - When a topology is shut down - this MUST be the last SDAM event fired.
    * - ``ServerHeartbeatStartedEvent``
-     - Published when the server monitor sends its ``hello`` or legacy hello call to the server.
+     - Published when the server monitor sends its ``hello`` or legacy hello call to the server. When the monitor is creating a new connection, this event MUST be published just after the socket is created and just before the ``hello`` or legacy hello.
```

Comment:
is this right?  I don't think this is what we do in Node.  we emit heartbeat started events before establishing a socket (before a call to `connect`)

Reply:
Shane and I are discussing this further, but our initial decision was to do what we currently do in the Node driver, but this does make it so that the the `duration` on the first `ServerHeartbeatSucceededEvent` would include the time taken to complete the TCP handshake which adds at least another network round trip, artificially inflating the RTT since that gets calculated from the aforementioned `duration` field.

How I currently have this written would avoid this, but would require changes in the Node driver among others.

Reply:
Update: Will change this to expect the `ServerHeartbeatStartedEvent` before connection creation. There was confusion on my end as I thought we had to measure the start time as when the `ServerHeartbeatStartedEvent` was emitted. On Node's end this will require still require us to change how we implement our monitoring since we currently measure the heartbeat duration starting as mentioned above.